woocommerce/wc-admin
# Filters: Fix regular filters not working

Simple filters weren't working on `main`.

![Screen Shot 2020-06-26 at 2 32 50 PM](https://user-images.githubusercontent.com/1922453/85814409-fb3af800-b7b9-11ea-8037-7543f82d559c.png)

https://github.com/woocommerce/woocommerce-admin/pull/4650 added properties to the `staticParams` config object such that certain parameters wouldn't change as the user navigates a report. These are mostly used for things like chart type and page. See https://github.com/woocommerce/woocommerce-admin/pull/4650#issuecomment-647336121 for previous discussion where filter parameters were added. This broke the basic filtering because those parameters were considered static.

### Detailed test instructions:

1. Product Report.
2. Filter by single product: Show > Single Product 
3. See the filter work (this is broken on `main`.

Also go through the tests of #4650 to make sure no regression occured.

- Open WooCommerce -> Customers
- Set up an advanced filter, make sure to press the "Filter" button to apply it
- Select the filter type dropdown and reselect "Advanced"
- Observe that the advanced filters remain
- Select the filter type dropdown and select a different filter type, then reselect Advanced
- Observe that the advanced filters remain
- Refresh the page
- Observe that the advanced filters remain
- There should be no JS errors in the console relating to missing `key` values
